### PR TITLE
fix(dropdown): Fix dropdown display issues.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** N/A

## Summary

fix(dropdown): Fix dropdown display issues.

### CODE REVIEW

File: src/components/calcite-dropdown/calcite-dropdown.tsx [hunk 1]
```diff
@@ -223,9 +223,13 @@ export class CalciteDropdown {
 
   @Listen("calciteDropdownOpen", { target: "window" })
   closeCalciteDropdownOnOpenEvent(e: Event): void {
-    if (e.target !== this.el) {
-      this.active = false;
+    const composedPath = e.composedPath();
+
+    if (composedPath.includes(this.el)) {
```
Comment (jcfranco) [inline src/components/calcite-dropdown/calcite-dropdown.tsx:228]: Nitpick: `e.composedPath().includes(this.el)` since the path is only referenced once.

**Review by jcfranco — APPROVED** [review]
👍🏼

**Comment by caripizza** [comment]
Thanks @driskull for fixing that split-button dropdown issue so quickly! Do you know of any way we can add test coverage for the fix or what are your thoughts on that?

**Comment by driskull** [comment]
I think we should have a story that shows the dropdown opened when its set to auto width. I think that would have caught this.

**Comment by driskull** [comment]
@jcfranco can you review again? I'd like to update the stories to have at least one dropdown open by default. This means adding a prop to the split-button.

File: src/components/calcite-split-button/calcite-split-button.tsx [hunk 2]
```diff
@@ -21,6 +21,12 @@ export class CalciteSplitButton {
   /** is the control disabled  */
   @Prop({ reflect: true }) disabled?: boolean;
 
+  /**
+   * Is the dropdown currently active or not
+   * @internal
```
Comment (driskull) [inline src/components/calcite-split-button/calcite-split-button.tsx:26]: I can make this public in a separate PR or we could do it at a later time.

Comment (jcfranco) [reply src/components/calcite-split-button/calcite-split-button.tsx:26]: 👍  to revisit making it public later.

**Review by jcfranco — APPROVED** [review]
👍🏼

**Comment by jcfranco** [comment]
> Do you know of any way we can add test coverage for the fix or what are your thoughts on that?

For anything that specifically needs to test `composedPath` in event handling, we'll need some sort of shadow DOM wrapper util. We can bring back the e2e test page setup util from `calcite-app-components` to make it easy to use.

**Pull request merged**